rules/xcodeproj.bzl

```diff
@@ -269,9 +278,22 @@ def _xcodeproj_impl(ctx):
             framework_search_paths.append("\"%s\"" % fi)
         target_settings["FRAMEWORK_SEARCH_PATHS"] = " ".join(framework_search_paths)
         target_settings["GCC_PREPROCESSOR_DEFINITIONS"] = " ".join(["\"%s\"" % d for d in target_info.cc_defines.to_list()])
+
         if target_info.product_type == "application":
             target_settings["INFOPLIST_FILE"] = "$BAZEL_STUBS_DIR/Info-stub.plist"
             target_settings["PRODUCT_BUNDLE_IDENTIFIER"] = target_info.bundle_id
+        else:
+            defines_without_equal_sign = []
+            for d in target_info.swift_defines.to_list():
+                splits = d.split("=")
+                append_this_define = True
+
+                # So that definition with value zero won't be populated
+                if len(splits) > 1 and splits[1] == "0":
```

Comment:
so we have Foo=0 defined, it won't show up on swift side at all

Reply:
See `swift/internal/compiling.bzl` in rules_swift -- this seems like a re-implementation of `_exclude_swift_incompatible_define`

Reply:
wow how did you find this method... indeed it is identical in terms of the goal of the method. 
however based off the documentation of its caller `_conditional_compilation_flag_configurator` it only adds the flags to command line (that's how we get swift compile correctly?) and for us it's to add the flags into build setting.
If I remove this check the test will fail because there is a `foo=0` flag being added as well. Therefore I just borrowed their code to make them consistent to each other. Is that an okay thing to do?